java/client/src/main/java/glide/api/RedisClient.java

```diff
@@ -272,6 +277,27 @@ public CompletableFuture<String> functionDelete(@NonNull String libName) {
                 FunctionDelete, new String[] {libName}, this::handleStringResponse);
     }
 
+    @Override
+    public CompletableFuture<byte[]> functionDump() {
+        return commandManager.submitNewCommand(
+                FunctionDump, new GlideString[0], response -> handleBytesResponse(response).getBytes());
```

Comment:
With the correct handler you can do `handleBytesResponse`
```suggestion
                FunctionDump, new GlideString[0], this::handleBytesResponse);
```